Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -260,17 +289,40 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         diagnosticEngine: DiagnosticEngine = .init(),
         configuration: Configuration = .init()
     ) throws {
-        self.dataProvider = dataProvider
+        self.dataProvider = .legacy(dataProvider)
         self.diagnosticEngine = diagnosticEngine
         self._configuration = configuration
         
-        self.dataProvider.delegate = self
+        _legacyDataProvider.delegate = self
         
         for bundle in dataProvider.bundles.values {
             try register(bundle)
         }
     }
-    
+
+    /// Initializes a documentation context with a given `bundle`.
+    ///
+    /// - Parameters:
+    ///   - bundle: The bundle to register with the context.
+    ///   - fileManager: The file manager that the context uses to read files from the bundle.
+    ///   - diagnosticEngine: The pre-configured engine that will collect problems encountered during compilation.
+    ///   - configuration: A collection of configuration for the created context.
+    /// - Throws: If an error is encountered while registering a documentation bundle.
+    package init(
```

Comment:
Why do we need to retain the old initializer? ...and add the complexity around:

```
 private enum _Provider {
        case legacy(DocumentationContextDataProvider)
        case new(DocumentationBundleDataProvider)
    }
```

Is the problem that the old initializer is public? And so we can't remove it?

Should you deprecate the old initializer so we can one day, after the next major release possibly, remove the old initializer and the legacy data provider?

Reply:
Correct. Because the old initializer is public we need to keep it around and retain support for one full _minor_ release. I didn't deprecate it in this PR because it's still used in tests so deprecating it would introduce warnings. #1059 deprecates it [here](https://github.com/swiftlang/swift-docc/pull/1059/files#diff-c452db10128fc43b2e77212a1a282fe0834e541e10852fa8dd50f7c9b480b4e7R292).